### PR TITLE
(It is just to fix/improve the process to generate auto PRs for weave)fix to allow generate diff templates for 2.6.x

### PR DESCRIPTION
**Pull request by camilamacedo86**
#### What this PR does / why we need it:

Fix AddOn automation after we apply a change valid only for 2.8+

### CODE REVIEW

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: addons/weave/template/generate.sh [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 set -euo pipefail
 
+# Source the functions file to semver
+source ./../../../scripts/common/common.sh
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply addons/weave/template/generate.sh:6]: What do you mean with that?
Why we would not able to use source?
Should better just copy and paste the implementation?

[the remark this answers is not in the document]
Comment (emosbaugh) [reply addons/weave/template/generate.sh:6]: could you just change it to this?

```
if [[ "$ADDON_VERSION" == 2.8.* ]]; then
```

File: addons/weave/template/generate.sh [hunk 2]
```diff
@@ -63,7 +63,12 @@ function generate() {
 
     if [ -n "$WEAVE_EXEC_IMAGE_PATCH_VERSION" ]; then
         sed -i "s/weaveworks\/weaveexec:$VERSION/kurlsh\/weaveexec:${WEAVE_EXEC_IMAGE_PATCH_VERSION}/" "$dir/Manifest"
-        sed -i "s/weaveworks\/weaveexec:$VERSION/kurlsh\/weaveexec:${WEAVE_EXEC_IMAGE_PATCH_VERSION}/" "$dir/patch-daemonset.yaml"
+
+        if [[ "$ADDON_VERSION" == 2.8.* ]]; then
+            sed -i "s/weaveworks\/weaveexec:$VERSION/kurlsh\/weaveexec:${WEAVE_EXEC_IMAGE_PATCH_VERSION}/" "$dir/patch-daemonset.yaml"
+        else
+            sed -i "s/weaveworks\/weaveexec:$VERSION/kurlsh\/weaveexec:${WEAVE_EXEC_IMAGE_PATCH_VERSION}/" "$dir/patch-daemonset_legacy.yaml"
```
Comment (emosbaugh) [inline addons/weave/template/generate.sh:70]: i think you will have to rename this file to patch-daemonset.yaml so its in the kustomization.yaml

**Review by emosbaugh — CHANGES_REQUESTED** [review]

File: addons/weave/template/generate.sh [hunk 3]
```diff
@@ -64,6 +67,16 @@ function generate() {
...
+
+        # The following code will remove the initContainer added to sort out issue face with weave and symbolic links
+        semverParse "${ADDON_VERSION}"
+        local addon_major_minor_version="${major}.${minor}"
+        if [ "$addon_major_minor_version" != "2.8" ]; then
+            # Remove initContainers section
+            sed -i '/initContainers:/,/volumes:/d' "$dir/patch-daemonset.yaml"
+            # Remove volumes section
+            sed -i '/volumes:/,+1d' "$dir/patch-daemonset.yaml"
+        fi
```
Comment (camilamacedo86) [inline addons/weave/template/generate.sh:79]: @emosbaugh I think I found a better way

Comment (emosbaugh) [reply addons/weave/template/generate.sh:76]: could you instead have two different files entirely with a patch for each version? this seems error prone

**Review by emosbaugh — CHANGES_REQUESTED** [review]

**Review by laverya — APPROVED** [review]

**Comment by laverya** [comment]
After this gets merged, we should run it to see what changes it makes for sure

**Comment by camilamacedo86** [comment]
Hi @laverya,

Thank you a lot. 
I did that now, see: https://github.com/replicatedhq/kURL/actions/runs/5318403531/jobs/9647260756. All fine.

**Pull request merged**